src/coreclr/jit/morphblock.cpp

```diff
@@ -144,7 +144,9 @@ GenTree* MorphInitBlockHelper::Morph()
     assert(m_result != nullptr);
 
 #ifdef DEBUG
-    if (m_result != m_asg)
+    // If we are going to return a different node than the input then morph
+    // expects us to have set GTF_DEBUG_NODE_MORPHED.
+    if ((m_result != m_asg) || (sideEffects != nullptr))
```

Comment:
Fix for a bug introduced in #83590 where we could hit asserts due to not marking some nodes with `GTF_DEBUG_NODE_MORPHED`.